ADQL.tex

```diff
@@ -2799,19 +2799,87 @@ \subsubsection{IN\_UNIT}
+
+The first argument MUST be a numeric expression.
 
 The second argument MUST be a string literal containing a valid unit
 description using the formatting defined in the \VOUnitSpec.
 
+It has to be noted that a unitless value, as defined by the \VOUnitSpec, can not
+be converted. Similarly a non-unitless value can not be converted into a
+unitless one.
+
 The system SHOULD report an error in response to the following conditions:
 \begin{itemize}
     \item if the second argument is not a valid unit description
-    \item if the system is not able to convert the value into the requested units.
+    \item if the system is not able to convert the value into the requested unit
 \end{itemize}
 
-Details of the mechanism for reporting the error condition are
-implementation dependent.
+Examples:
+
+\begin{verbatim}
+    -- ra       : column expressed in 'deg'
+    -- phot_cnt : column expressing a unitless quantity
+
+    IN_UNIT(ra, '')          -- ERROR: Conversion into a 'unitless'
+                             --        value impossible
+
+    IN_UNIT(phot_cnt, 'deg') -- ERROR: Impossible to convert a
+                             --        unitless value
+
+    IN_UNIT(ra, 'foo')       -- ERROR: Incorrect unit definition
+
+    IN_UNIT(ra, 'kg')        -- ERROR: Impossible conversion
+                             --        (deg -> kg)
+\end{verbatim}
+
+Details of the mechanism for reporting the error condition are implementation
+dependent.
+
+Currently, ADQL does not provide any way to specify the unit of a numeric
+literal (e.g. a number such as \verb:42: or \verb:3.14:) or of the result of an
+operation or function. Consequently, in regards of the unitless aspect of such
+numeric expression, \verb:IN_UNIT(): SHOULD return an error.
+
+Examples:
+
+\begin{verbatim}
+    IN_UNIT(42, 'kg')        -- ERROR: no unit can be specified for
+                             --        42, so, it is unitless
+                             --        => its conversion is
+                             --           impossible
+
+    IN_UNIT(40+2, 'deg')     -- ERROR: idem
+
+    IN_UNIT(sqrt(42), 'deg') -- ERROR: idem
+\end{verbatim}
+
+An ADQL service implementation could use a unit inference mechanism. In such
+case, unit of relatively complex numeric expressions could be guessed. With the
+help of a such mechanism, \verb:IN_UNIT(): MAY convert more complex
+expressions.
+
+Examples, with an \emph{advanced} unit inference mechanism:
+
+\begin{verbatim}
+    -- ra       : column expressed in 'deg'
+    -- ra_error : column expressed in 'deg'
+    -- pm_ra    : column expressing in 'deg/yr'
+    -- pm_dec   : column expressing in 'deg/yr'
+
+    IN_UNIT(ra+ra_error, 'rad')                             -- OK
+
+    IN_UNIT(ra+10, 'rad')                                   -- OK
+
+    IN_UNIT(sqrt(power(pm_ra,2)+power(pm_dec,2)), 'rad/yr') -- OK
+
+    IN_UNIT(ra + IN_UNIT(10, 'deg'), 'rad')                 -- OK
```

Comment:
Sorry - why is this OK ?
If `IN_UNIT(42, 'kg')` generates an error because `42` is unitless, then why doesn't `IN_UNIT(10, 'deg')` raise a similar error ?

Reply:
Yes, you're right. It is a left-over of another aborted version :(
Coming soon a new PR to fix it.
Thank you for having noticed it :-)